deadlines/src/main/java/com/palantir/deadlines/ExpirationObservation.java

```diff
@@ -0,0 +1,174 @@
+import com.google.common.annotations.VisibleForTesting;
+import com.palantir.deadlines.DeadlineMetrics.Expired_Cause;
+import com.palantir.deadlines.DeadlineMetrics.Expired_Intent;
+import com.palantir.tritium.metrics.registry.SharedTaggedMetricRegistries;
+
+/**
+ * Provides the ability to observe the number of deadline expirations that happen between points in time.
+ * <p/>
+ * Usage example:
+ * <pre>
+ *     ExpirationObservation start = ExpirationObservation.start();
+ *     // ... operation that may cause deadline expiration ...
+ *     ExpirationObservation end = start.observeFrom();
+ *     if (end.totalExpirations() > 0) {
+ *         // at least one deadline expiration happened
+ *     }
+ * </pre>
+ *
+ * Calling {@link #start()} will create a new observation with no recorded expirations.
+ *
+ * Calling {@link #observeFrom()} will return an observation comparing the current point in time to the count of
+ * expirations when the original observation was created.
+ *
+ * Calling {@link #totalExpirations()} will return the number of deadline expirations that occurred
+ * between two observations.
+ */
+public final class ExpirationObservation {
+
+    private static final DeadlineMetrics metrics = DeadlineMetrics.of(SharedTaggedMetricRegistries.getSingleton());
+
+    private final MeterValues start;
+    private final MeterValues end;
+
+    private ExpirationObservation(MeterValues start, MeterValues end) {
+        this.start = start;
+        this.end = end;
+    }
+
+    /**
+     * Begin a new observation.
+     */
+    public static ExpirationObservation start() {
+        MeterValues now = readMeters();
+        return new ExpirationObservation(now, now);
+    }
+
+    /**
+     * Return a new observation with the number of deadline expirations that have occurred since this observation was
+     * initialized.
+     *
+     * The current observation should have been previously initialized either via a call to this method, or
+     * a call to {@link #start()},
+     */
+    public ExpirationObservation observeFrom() {
+        MeterValues now = readMeters();
+        return new ExpirationObservation(end, now);
+    }
+
+    /**
+     * Return the total number of expirations that have occurred since this observation was initialized.
+     */
+    public long totalExpirations() {
+        return nExternalPropagate()
+                + nExternalPropagateAlreadyExpired()
+                + nExternalIgnore()
+                + nInternalPropagate()
+                + nInternalPropagateAlreadyExpired()
+                + nInternalIgnore();
+    }
+
+    @VisibleForTesting
+    long totalWithExternalCause() {
+        return nExternalPropagate() + nExternalPropagateAlreadyExpired() + nExternalIgnore();
+    }
+
+    @VisibleForTesting
+    long totalWithInternalCause() {
+        return nInternalPropagate() + nInternalPropagateAlreadyExpired() + nInternalIgnore();
+    }
+
+    @VisibleForTesting
+    long totalWithPropagateIntent() {
```

Comment:
we could make this (or all of these) public to allow consumers to e.g. observe when a deadline expiration would throw an exception if enforcement is enabled.

Reply:
What are your thoughts on something like #95? I think it's a bit simpler: I'm proposing we just expose the deadline metric values and provide a small utility to compute diffs between the metric value observations.